src/modules/launcher/Wox.Plugin/Common/ShellLocalization.cs

```diff
@@ -0,0 +1,93 @@
+
+        [DllImport("shell32.dll", CallingConvention = CallingConvention.Winapi, CharSet = CharSet.Unicode)]
+        internal static extern int SHGetLocalizedName(string pszPath, StringBuilder pszResModule, ref int cch, out int pidsRes);
+
+        [DllImport("user32.dll", EntryPoint = "LoadStringW", CallingConvention = CallingConvention.Winapi, CharSet = CharSet.Unicode)]
+        internal static extern int LoadString(IntPtr hModule, int resourceID, StringBuilder resourceValue, int len);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode, ExactSpelling = true, EntryPoint = "LoadLibraryExW")]
+        internal static extern IntPtr LoadLibraryEx(string lpFileName, IntPtr hFile, uint dwFlags);
+
+        [DllImport("kernel32.dll", ExactSpelling = true)]
+        internal static extern int FreeLibrary(IntPtr hModule);
+
+        [DllImport("kernel32.dll", EntryPoint = "ExpandEnvironmentStringsW", CharSet = CharSet.Unicode, ExactSpelling = true)]
+        internal static extern uint ExpandEnvironmentStrings(string lpSrc, StringBuilder lpDst, int nSize);
+
+        /// <summary>
+        /// Returns the localized name of a shell item.
+        /// </summary>
+        /// <param name="path">Path to the shell item (e. g. shortcut 'File Explorer.lnk').</param>
+        /// <returns>The localized name as string or <see cref="string.Empty"/>.</returns>
+        public static string GetLocalizedName(string path)
+        {
+            StringBuilder resourcePath = new StringBuilder(1024);
+            StringBuilder localizedName = new StringBuilder(1024);
+            int len, id;
+            len = resourcePath.Capacity;
+
+            // If there is no resource to localize a file name the method returns a non zero value.
+            if (SHGetLocalizedName(path, resourcePath, ref len, out id) == 0)
+            {
+                _ = ExpandEnvironmentStrings(resourcePath.ToString(), resourcePath, resourcePath.Capacity);
+                IntPtr hMod = LoadLibraryEx(resourcePath.ToString(), IntPtr.Zero, DONTRESOLVEDLLREFERENCES | LOADLIBRARYASDATAFILE);
+                if (hMod != IntPtr.Zero)
+                {
+                    if (LoadString(hMod, id, localizedName, localizedName.Capacity) != 0)
+                    {
+                        string lString = localizedName.ToString();
+                        _ = FreeLibrary(hMod);
+                        return lString;
+                    }
+
+                    _ = FreeLibrary(hMod);
+                }
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// This method returns the localized path to a shell item (folder or file)
+        /// </summary>
+        /// <param name="path">The path to localize</param>
+        /// <returns>The localized path or the original path if localized version is not available</returns>
+        public static string GetLocalizedPath(string path)
```

Comment:
not used.. Is needed?

Reply:
Currently not used. This is for later usage. (For example in folder plugin.)

And maybe code can be improved later by adding cache for already localized shell items.

Reply:
okay.. are you willing to try the Folder localization ? :) This is good to go 🥳

Reply:
These code is tested and works. But I didn't implement it in the existing plugins because of the time delay. That's why I suggested the cache in my last comment.

If you want me to work on this you can open an issue and assign me. (Important: Please add a note to the issue description that we should implement the caching feature in the `GetLocalizedName()` method to cache every translated item.)

Reply:
Created #19521 and assigned to you :)